RecoMuon/TrackerSeedGenerator/src/SeedMvaEstimator.cc

```diff
@@ -0,0 +1,119 @@
+#include "RecoMuon/TrackerSeedGenerator/interface/SeedMvaEstimator.h"
+
+#include "DataFormats/Math/interface/deltaPhi.h"
+#include "DataFormats/Math/interface/deltaR.h"
+
+#include "FWCore/ParameterSet/interface/FileInPath.h"
+#include "CommonTools/MVAUtils/interface/GBRForestTools.h"
+
+SeedMvaEstimator::SeedMvaEstimator(const edm::FileInPath& weightsfile,
+                                   const std::vector<double>& scale_mean,
+                                   const std::vector<double>& scale_std) {
+  gbrForest_ = createGBRForest(weightsfile);
```

Comment:
Thanks for the comments.
Could you let me know how to estimate the memory needed by GBRForest? Is there simple way or function?

Reply:
> Thanks for the comments.
> Could you let me know how to estimate the memory needed by GBRForest? Is there simple way or function?

You can use the igprof profiler.
Please have a look at https://twiki.cern.ch/twiki/bin/viewauth/CMS/RecoIntegration#Run_profiler_igprof

Reply:
I got several files after running commands in link above. I not sure what numbers represent the memory needed by GBRForest, but I found these lines in a ig.1.txt
```
INSERT INTO symbols VALUES (585450176, "createGBRForest(edm::FileInPath const&)", 585378480);
INSERT INTO mainrows VALUES (1164, 585450176, 0, 3528231, 3528231, 0, 72024, 3, 3,    0.32);
INSERT INTO parents VALUES (1162, 1164, 3528231, 72024, 3,    0.32);
INSERT INTO children VALUES(1165, 1164, 3528231, 72024, 3,    0.32);
INSERT INTO symbols VALUES (585253744, "createGBRForest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > >&)", 585378480);
```
If you are more convenient with checking these directly, these files(config, its log, ig.1.sql3(txt), or IgProf.1(250).gz) are uploaded in lxplus ( /afs/cern.ch/user/w/wjun/public/)

Reply:
@makortel , according to the igProf output produced by @wonpoint4 this is the LIVE_MEM taken by GBRForest:

```
[1164]      0.3      3'528'231              0 / 3'528'231             72'024                  createGBRForest(edm::FileInPath const&)
            0.3  .............      3'528'231 / 3'528'231             72'024 / 72'024           createGBRForest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > >&) [1165]
```

At a first glance, this is a small fraction of the total of the job (which I imagine is the test HLT run described in the recipe for testing the code).

Do you think it is manageable as such, or would you suggest  storing the SeedMvaEstimator in a GlobalCache instead? (In that case, I think @wonpoint4 will need some suggestion for the implementation)

Reply:
To me 3.4 MB/stream sounds like something that could be worth of addressing (although I don't have good feeling on how tight the memory is in HLT). I can cook up an example on GlobalCache, and I don't think it has to necessarily hold this PR.

Reply:
> (although I don't have good feeling on how tight the memory is in HLT)

Not particularly tight, but of course the less the better :-)
A GlobalCache approach should not have any (negative) impact on the runtime performance, should it ?

Reply:
> A GlobalCache approach should not have any (negative) impact on the runtime performance, should it ?

It indeed should not (since `GBRForest` is thread safe and lock free).